src/main/java/net/whistlingfish/harmony/HarmonyClient.java

```diff
@@ -114,7 +126,39 @@ public void connect(String host, String username, String password) {
             connection.connect();
             connection.login(oaResponse.getUsername(), oaResponse.getPassword(), "main");
             connection.setFromMode(FromMode.USER);
-
+            connection.addConnectionListener(new ConnectionListener() {
+				
+				@Override
+				public void reconnectionSuccessful() {
+					getCurrentActivity();
+				}
+				
+				@Override
+				public void connected(XMPPConnection connection) {
+				}
+
+				@Override
+				public void authenticated(XMPPConnection connection) {
+				}
+
+				@Override
+				public void connectionClosed() {
+				}
+
+				@Override
+				public void connectionClosedOnError(Exception e) {
+				}
+
+				@Override
+				public void reconnectingIn(int seconds) {
+				}
+
+				@Override
+				public void reconnectionFailed(Exception e) {
+				}
+				
+			});
+            
```

Comment:
If the underlying connection drops we should update the current state when it comes back